cirq/sim/abstract_state.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+from typing import Any
+
+
+class AbstractState(metaclass=abc.ABCMeta):

```

Comment:
The [nomenclature docs](https://github.com/quantumlib/Cirq/blob/master/docs/dev/nomenclature.md) caution against using `state` as a type name, since it's a heavily overloaded term. That being said, whether this is a density matrix or state vector depends on the implementation; `state_rep` seems like the closest analog in the doc. Maybe `AbstractStateRepresentation` for this?